mesonbuild/compilers/compilers.py

```diff
@@ -359,6 +360,11 @@ def get_base_compile_args(options: 'KeyedOptionDictType', compiler: 'Compiler')
             args.extend(compiler.get_profile_use_args())
     except KeyError:
         pass
+    try:
+        if options[OptionKey('b_profile')].value:
+            args += compiler.get_gprof_args()
```

Comment:
Given the general name (b_profile), shouldn't the function be named more generically as well?